backend/tests/test_chmeme_rec.py

```diff
@@ -0,0 +1,95 @@
+        self.rec_nums = list(data.keys())
+        keys = list(data[self.rec_nums[0]].keys())
+        self.key_len = len(keys)
+        if embedding_model_name == 'online':
+            self.embedding_model_url = os.getenv("EMBEDDING_URL")
+            if self.embedding_model_url is None:
+                self.embedding_model_url = "http://103.177.28.193:30086/embedding"
+        elif embedding_model_name == 'jina':
+            self.embedding_model = AutoModel.from_pretrained("jinaai/jina-embeddings-v3", trust_remote_code=True)
+        else:
+            raise KeyError(f'Incorrect arg: {self.embedding_model_name}')
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        if self.embedding_model_name != 'online':
+            self.embedding_model = self.embedding_model.to(device)
+
+    def _get_embed_sim(self, query: str):
+        data_sentence = []
+        for num in self.rec_nums:
+            item = self.data[num]
+            for key in list(item.keys()):
+                data_sentence.append(item[key])
+        if self.embedding_model_name == 'bert':
+            similarities = self.embedding_model.similarity(query, data_sentence)
+        else:
+            if self.embedding_model_name == 'online':
+                payload = {
+                    "text": [query] + data_sentence,
+                    # "text": ['你好','你好好'],
+                    "model": "bge-large-zh-v1.5"
+                }
+                headers = {"Content-Type": "application/json"}
+                response = requests.post(self.embedding_model_url, json=payload, headers=headers)
+                embeddings = response.json()['embedding']
+            else:
+                embeddings = self.embedding_model.encode([query] + data_sentence)
+            embeddings_query = torch.tensor(embeddings[0]).reshape(1, -1)
+            embeddings_sentences = torch.tensor(embeddings[1:])
+            similarities = cosine_similarity(embeddings_query, embeddings_sentences)
+        return similarities[0]
+
+    def _get_sim_score(self, query: str):
+        sims = self._get_embed_sim(query)
+        sims = torch.from_numpy(sims)
+        key_len = self.key_len
+        sims = sims.view(-1, key_len).float()
+        scores = sims.mean(dim=1).tolist()
+        return scores
+
+    def get_topk_meme(self, query: str, k: int = 2) -> list:
+        scores = self._get_sim_score(query)
+        sorted_with_index = sorted(enumerate(scores), key=lambda x: x[1], reverse=True)
+
+        top_k_indexes = [index for index, _ in sorted_with_index[:k]]
+
+        res = []
+        for ind in top_k_indexes:
+            res.append(self.rec_nums[ind])
+        return res
+
+
+def test_rec_chmeme(query: str = '耗子尾汁', k: int = 4):
+    # json file path
+    #file_path = '/mnt/afs/wangqijian/meme/ch_memes/new_100_chmeme.json'
+    file_path = os.environ.get("file_path")
+    with open(file_path, 'r', encoding='utf-8') as file:
+        ch_dataset = json.load(file)
```

Comment:
dataset should already contain the complete path, you don't need to pack the path after the recommendation